javatests/build/bazel/tests/integration/WorkspaceDriverIntegrationTest.java

```diff
@@ -53,6 +53,91 @@ public void testRunningWithEnvironment() throws Exception {
     assertEquals("bazel test environment variable return code", 0, returnCode);
   }
 
+  /**
+   * Test that {@code junit} can be specified as a {@code java_import_external} without any download
+   * taking place. Indeed, because the cache is frozen (see {@link RepositoryCache#freeze()}), it is
+   * not possible to add any other dependency.
+   */
+  @Test
+  public void testExternalDep() throws Exception {
+    String testName = "TestMe";
+
+    driver.scratchFile(
+        "WORKSPACE",
+        "load(\"@bazel_tools//tools/build_defs/repo:java.bzl\", \"java_import_external\")",
+        "java_import_external(",
+        "    name = \"org_junit\",",
+        "    licenses = [\"restricted\"],  # Eclipse Public License 1.0",
+        "    jar_urls = [",
+        "        \"http://repo1.maven.org/maven2/junit/junit/4.11/junit-4.11.jar\",",
+        "    ],",
+        "    jar_sha256 = \"90a8e1603eeca48e7e879f3afbc9560715322985f39a274f6f6070b43f9d06fe\",",
+        ")");
+
+    driver.scratchFile("BUILD.bazel", testLabelNamed(testName));
+    driver.scratchFile("TestMe.java", passingTestNamed(testName));
+
+    Command cmd = driver.bazelCommand("test", "//:TestMe").build();
+
+    int returnCode = cmd.run();
+    assertEquals(0, returnCode);
+  }
+
+  /** Try (unsuccessfully) to download a dependency that was not specified as an external dep. */
+  @Test
+  public void testRepositoryCacheIsFrozen() throws Exception {
```

Comment:
@hchauvin It looks to me like there's a bit of a mixup (though I might be wrong).
1. Why is the second URL of junit-4.11? I think this works but is confusing.
2. Why is jcommander specified here? Doesn't it belong in the happy path above to see that it can be used (the other entry from the dict)?